wasm/jit/jit_amd64.go

```diff
@@ -5220,22 +5148,16 @@ func (c *amd64Compiler) callGoFunction(jitStatus jitCallStatusCode, addr wasm.Fu
 	// Emit the return instruction.
 	c.exit(jitStatus)
 
-	nopAfterReturnInst := c.newProg()
-	nopAfterReturnInst.As = obj.ANOP
-	c.addInstruction(nopAfterReturnInst)
-
-	readInstructionAddressCompletionCallback(moveRIP, nopAfterReturnInst)
-
 	// They were temporarily used, so we mark them unused.
 	c.locationStack.markRegisterUnused(regs...)
 	return nil
 }
 
-// readInstructionAddress adds the instruction to read the target instruction's absolute address into the destinationRegister.
-// At the callsite, the returned callback must be called with two arguments:
-// instructionAfterReadInstruction is the instruction added *right after* calling this readInstructionAddress function.
-// addressAcquisitionTargetInstruction  is the instruction we want to get the absolute address.
-func (c *amd64Compiler) readInstructionAddress(destinationRegister int16) func(instructionAfterReadInstruction, addressAcquisitionTargetInstruction *obj.Prog) {
+// readInstructionAddress add a LEA instruction to read the target instruction's absolute address into the destinationRegister.
+// beforeAcquisitionTargetInstruction is the instruction kind (e.g. RET, JMP, etc.) right before the instruction
+// of which the callsite wants to aquire the absolute address.
+func (c *amd64Compiler) readInstructionAddress(destinationRegister int16, beforeAcquisitionTargetInstruction obj.As) {
```

Comment:
really simplified this and now this doesn't return callback

Reply:
maybe now I can write a simple test on this lemme

Reply:
added https://github.com/tetratelabs/wazero/pull/175/commits/af1d400a51331bf708a01d7c0496b8bd27b80bee